src/map/status.c

```diff
@@ -3479,31 +3479,31 @@ static void status_calc_regen_rate(struct block_list *bl, struct regen_data *reg
 	}
 
 
+	if (sc->data[SC_MAGNIFICAT]) {
+#ifndef RENEWAL // HP Regen applies only in Pre-renewal
+		regen->rate.hp += regen->rate.hp * 100 / 100;
+#endif
+		regen->rate.sp += regen->rate.sp * 100 / 100;
+	}
+
 	// Tension relax allows the user to recover HP while overweight
 	// at 1x speed. Other SC ignored? [csnv]
 	if (sc->data[SC_TENSIONRELAX]) {
 		if (sc->data[SC_WEIGHTOVER50] || sc->data[SC_WEIGHTOVER90]) {
 			regen->flag &= ~RGN_SP;
-			regen->rate.hp = 100;
+			regen->rate.hp += regen->rate.hp * 100 / 100;
```

Comment:
Isn't this redundant * 100 / 100

Reply:
it is, i just kept it this way so it's clear it's a cumulative ratio.